mesmer/core/datatree.py

```diff
@@ -0,0 +1,156 @@
+import xarray as xr
+from datatree import DataTree
+
+
+def _extract_single_dataarray_from_dt(dt: DataTree) -> xr.DataArray:
+    """
+    Extract a single DataArray from a DataTree node, holding a ``Dataset`` with one ``DataArray``.
+    """
+    if not dt.has_data:
+        raise ValueError("DataTree must contain data.")
+
+    ds = dt.to_dataset()
+    name, *others = ds.keys()
+    if others:
+        raise ValueError("DataTree must only contain one data variable.")
+
+    da = ds[name]
+    return da
+
+
+def collapse_datatree_into_dataset(
+    dt: DataTree, dim: str, **concat_kwargs
+) -> xr.Dataset:
+    """
+    Take a ``DataTree`` and collapse **all subtrees** in it into a single ``xr.Dataset`` along dim.
+    All subtrees are converted to ``xr.Dataset`` objects and concatenated along the
+    specified dimension. The dimension along which the datasets are concatenated will be added as a
+    coordinate to the resulting dataset and name of each subtree will be used as the coordinate
+    values for this new dimension. Internally, xr.concat is used to concatenate the datasets, so
+    all keyword arguments that can be passed to xr.concat can be passed to this function as well.
+
+    Parameters
+    ----------
+    dt : DataTree
+        The DataTree to collapse.
+    dim : str
+        The dimension to concatenate the datasets along.
+    **concat_kwargs : dict
+        Additional keyword arguments to pass to ``xr.concat``.
+
+    Returns
+    -------
+    xr.Dataset
+        The collapsed dataset.
+    """
+    # TODO: could potentially be replaced by DataTree.merge_child_nodes in the future?
+    datasets = [subtree.to_dataset() for subtree in dt.subtree if not subtree.is_empty]
+
+    # Concatenate datasets along the specified dimension
+    ds = xr.concat(datasets, dim=dim, **concat_kwargs)
+    dims = [subtree.name for subtree in dt.subtree if not subtree.is_empty]
+    ds = ds.assign_coords({dim: dims})
+
+    return ds
+
+
+def stack_linear_regression_datatrees(
```

Comment:
```suggestion
def stack_datatrees_linear_regression(
```

?

or `stack_datatrees_for_linear_regression`